### PR TITLE
MegEngine support

### PR DESCRIPTION
**Pull request by wangx26**
MegEngine is a fast, scalable and easy-to-use deep learning framework, with auto-differentiation.

[MegEngine](https://megengine.org.cn/)
[MegEngine github](https://github.com/MegEngine/MegEngine/)

For this pull request, i try to integrate MegEngine functionalities into netron.

#607

### CODE REVIEW

File: source/index.html [hunk 1]
```diff
@@ -295,7 +295,7 @@
     <button id="consent-accept-button" class="center consent-accept-button">Accept</button>
     <button id="open-file-button" class="center open-file-button">Open Model&hellip;</button>
     <button id="download-button" class="center download-button">Download App</button>
-    <input type="file" id="open-file-dialog" class="open-file-dialog" multiple="false" accept=".onnx, .pb, .meta, .tflite, .lite, .tfl, .keras, .h5, .hd5, .hdf5, .json, .model, .mar, .params, .param, .armnn, .mnn, .ncnn, .tnnproto, .nn, .uff, .paddle, .pdmodel, .dnn, .cmf, .mlmodel, .caffemodel, .pbtxt, .prototxt, .pkl, .pt, .pth, .t7, .joblib, .cfg, .xml">
+    <input type="file" id="open-file-dialog" class="open-file-dialog" multiple="false" accept=".onnx, .pb, .meta, .tflite, .lite, .tfl, .keras, .h5, .hd5, .hdf5, .json, .model, .mar, .params, .param, .armnn, .mnn, .ncnn, .tnnproto, .nn, .uff, .paddle, .pdmodel, .dnn, .cmf, .mlmodel, .caffemodel, .pbtxt, .prototxt, .pkl, .pt, .pth, .mge, .t7, .joblib, .cfg, .xml">
```
Comment (lutzroeder) [inline source/index.html:298]: Add to `app.js` and `electron-builder.yml` as well.

File: source/mge-metadata.json [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+        { "name": "num_parameters", "default": 1 },
+        { "name": "init", "default": 0.25 }
+      ],
+      "category": "Activation"
+    }
+  },
+  {
+    "name": "megengine.module.activation.ReLU",
+    "schema": {
+      "attributes": [],
```
Comment (lutzroeder) [inline source/mge-metadata.json:24]: Remove?

File: source/mge-pickle.js [hunk 3]
```diff
@@ -0,0 +1,563 @@
+/* jshint esversion: 6 */
+
+var pickle = pickle || {};
+
+pickle.Unpickler = class {
```
Comment (lutzroeder) [inline source/mge-pickle.js:5]: Remove duplicate pickle reader and use the one in `pickle.js`.

File: source/mge.js [hunk 4]
```diff
@@ -0,0 +1,1806 @@
...
+            }
+        }
+        
+    }
+
+    _loadModule(metadata, parent, groups, inputs) {
+       
+        for (const module of Object.keys(parent)){
+            const descriptor = Object.getOwnPropertyDescriptor(parent, module);
+            if(descriptor.value.__module__===undefined){
```
Comment (lutzroeder) [inline source/mge.js:106]: formatting

**Review by lutzroeder — CHANGES_REQUESTED** [review]
See below.

File: source/mge-pickle.js (hunk 3 above)
Comment (wangx26) [reply source/mge-pickle.js:5]: Line 239 in pickle.js don't support the Megengine files, so I have to copy a old version.
```
 else if (ArrayBuffer.isView(state) || Object(state) !== state) {
                        throw new pickle.Error('Invalid state dict' + (obj && obj.__module__ && obj.__name__ ? " for '" + obj.__module__ + '.' + obj.__name__ + "'": '') + '.');
                    }
```

**Comment by wangx26** [comment]
MegEngine  is a new open source deep learning framework,still being promoted and improved.
[shufflenet_v2_x1_0.zip](https://github.com/lutzroeder/netron/files/5424141/shufflenet_v2_x1_0.zip)
[shufflenet_v2_x1_0_state_dict.zip](https://github.com/lutzroeder/netron/files/5424143/shufflenet_v2_x1_0_state_dict.zip)

File: source/mge-pickle.js (hunk 3 above)
Comment (lutzroeder) [reply source/mge-pickle.js:5]: Can you share a repro and fix `pickle.js` to work for all scenarios.

**Review by lutzroeder — COMMENTED** [review]
Please address these issues or provide answers to these questions:

- [ ] Why is another custom format needed instead of MegEnging using an existing one like ONNX? There is almost no usage of this format.
- [ ] Run `make lint` and fix errors
- [ ] Most code in `mge.js` is copied and not needed for `.mge`. Please clean up and remove.
- [ ] Please share a repro and fix pickle.js to work for all scenarios so no duplicate code is needed.
